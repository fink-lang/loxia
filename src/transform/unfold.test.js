import {fink2js} from '../testing';


test('compiles unfold', ()=> {
  expect(
    fink2js(`
      unfold curr=start:
        (curr + inc, curr * 2)

      count = (start=0, inc=1):
        unfold curr=start:
          curr + inc
    `)
  ).toMatchSnapshot();
});
