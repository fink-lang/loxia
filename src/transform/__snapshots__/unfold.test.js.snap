// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiles unfold 1`] = `
"(function* unfold(curr = start) {
  while (true) {
    const $result1 = curr + inc;
    if ($result1 === stop) return;
    if ($result1 !== skip) yield $result1;
    curr = curr * 2;
  }
})();

const count = (start = 0, inc = 1) => function* unfold(curr = start) {
  while (true) {
    const $result2 = curr + inc;
    if ($result2 === stop) return;
    if ($result2 !== skip) yield $result2;
  }
}();"
`;
